shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Mempool.hs

```diff
@@ -54,13 +59,35 @@ import Control.State.Transition.Extended
 import Data.Functor ((<&>))
 import Data.Sequence (Seq)
 import Data.Typeable (Typeable)
+import NoThunks.Class (NoThunks)
 import Shelley.Spec.Ledger.API.Protocol (PraosCrypto)
 import Shelley.Spec.Ledger.LedgerState (NewEpochState)
 import qualified Shelley.Spec.Ledger.LedgerState as LedgerState
 import Shelley.Spec.Ledger.PParams (PParams' (..))
 import Shelley.Spec.Ledger.STS.Ledger (LedgerEnv, LedgerPredicateFailure)
 import qualified Shelley.Spec.Ledger.STS.Ledger as Ledger
 
+-- | A newtype which indicates that a transaction has been validated against
+-- some chain state.
+newtype Validated tx = Validated tx
+  deriving (Eq, NoThunks, Show)
+
+-- | Extract the underlying unvalidated Tx.
+extractTx :: Validated tx -> tx
+extractTx (Validated tx) = tx
+
+-- | Translate a validated transaction across eras.
+--
+-- This is not a `TranslateEra` instance since `Validated` is not itself
+-- era-parametrised.
+translateValidated ::
```

Comment:
I don't think this signature unblocks me.

I have a `Validated (Core.Tx (MaryEra c))` (where `Core.Tx` is a type family that reduces in this case to `SL.Tx (MaryEra c)`) but I need to invoked the following instance.

```
newtype Tx era = Tx {unTx :: Core.Tx era}

instance
  ( Crypto c,
    Core.Tx (AlonzoEra c) ~ ValidatedTx (AlonzoEra c)
  ) =>
  TranslateEra (AlonzoEra c) Tx
```

So the arguments to `Validated` are unequal and I have no way to line them up before invoking `translateValidated`.

Am I missing something? Thanks.

Reply:
I've added a `coerceValidated`, I _think_ this should fix stuff by allowing you to wrap/unwrap the newtypes inside. Otherwise let's just export the constructor and be done with it...

Reply:
That works! Thanks.